### PR TITLE
OneSignal integration

### PR DESCRIPTION
**Pull request by sadatetchamouza**
- Updated every deps for firebase and had to update the code due to some changes in the firebase usage

- Integrate oneSignal and test the push notification on Android.

### CODE REVIEW

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -36,7 +38,7 @@
 	<key>NSCameraUsageDescription</key>
 	<string>Can I use the camera please?</string>
 	<key>NSMicrophoneUsageDescription</key>
-    <string>Can I use the mic please?</string>
+	<string>Can I use the mic please?</string>
```
Comment (rygelouv) [inline ios/Runner/Info.plist:41]: Remind me to provide a better text for this before the release

File: lib/main.dart [hunk 2]
```diff
@@ -11,13 +13,30 @@ import 'screens.dart' show HomeScreen, LoginScreen, NoteEditor, SettingsScreen;
...
+  await Firebase.initializeApp();
+  initPlatformState();
+  runApp(NotesApp());
+}
+
+Future<void> initPlatformState() async {
+  //Remove this method to stop OneSignal Debugging
+  OneSignal.shared.setLogLevel(OSLogLevel.verbose, OSLogLevel.none);
+
+  OneSignal.shared.setAppId("00e4ca93-54c7-4c34-b9d4-ba4195bb36da");
```
Comment (rygelouv) [inline lib/main.dart:27]: This is ok for now, but we must find a way to secure this key.

File: pubspec.yaml [hunk 3]
```diff
@@ -29,22 +29,23 @@ dependencies:
...
+  intl: ^0.17.0
+  url_launcher: ^6.0.12
+  path_provider: ^2.0.5
   google_ml_kit: ^0.7.0
   camera: ^0.8.1+7
   image_cropper: ^1.2.3
   flutter_barcode_scanner: ^2.0.0
-  http: ^0.12.0
+  http: ^0.13.0
+  onesignal_flutter: ^3.2.3
```
Comment (rygelouv) [inline pubspec.yaml:49]: Thanks for updating these dependencies 👍

File: ios/OneSignalNotificationServiceExtension/NotificationService.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    
+    override func serviceExtensionTimeWillExpire() {
+        // Called just before the extension will be terminated by the system.
+        // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
+        if let contentHandler = contentHandler, let bestAttemptContent =  bestAttemptContent {
+            OneSignal.serviceExtensionTimeWillExpireRequest(self.receivedRequest, with: self.bestAttemptContent)
+            contentHandler(bestAttemptContent)
+        }
+    }
```
Comment (rygelouv) [inline ios/OneSignalNotificationServiceExtension/NotificationService.swift:40]: Nice swift code 😉

**Review by rygelouv — APPROVED** [review]

File: ios/Runner/Info.plist (hunk 1 above)
Comment (sadatetchamouza) [reply ios/Runner/Info.plist:41]: Ok noted

File: lib/main.dart (hunk 2 above)
Comment (sadatetchamouza) [reply lib/main.dart:27]: Ok I will read on how to store sensitive information in Flutter app to better store it

File: pubspec.yaml (hunk 3 above)
Comment (sadatetchamouza) [reply pubspec.yaml:49]: Yeah I hadn't the choice and also this make realize that there was lot of changes in the Firebase deps methods

File: ios/OneSignalNotificationServiceExtension/NotificationService.swift (hunk 4 above)
Comment (sadatetchamouza) [reply ios/OneSignalNotificationServiceExtension/NotificationService.swift:40]: Thanks, that's was provided in the OneSignal integration documentation

**Pull request merged**